packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -998,7 +1001,32 @@ class HeroController extends NavigatorObserver {
     BuildContext toHeroContext,
   ) {
     final Hero toHero = toHeroContext.widget as Hero;
-    return toHero.child;
+    if (MediaQuery.maybeOf(toHeroContext) == null && MediaQuery.maybeOf(fromHeroContext) == null) {
+      return toHero.child;
+    }
+    final MediaQueryData chosenMediaQueryData = MediaQuery.maybeOf(toHeroContext) ?? MediaQuery.of(fromHeroContext);
```

Comment:
If `MediaQuery.maybeOf(toHeroContext)` is null then I feel we don't want to fallback to `MediaQuery.of(fromHeroContext)`?

Reply:
Yeah maybe we just don't do transition at all?

Reply:
Yes, I think skipping the transition when either MediaQuery is null might be the way to go.